### ISSUE TITLE
Missing error for nested partially unknown types?

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#2942]
**Describe the bug**
Usually pyright emits an error for partially unknown / generic types. For nested types this sometimes doesn't happen.

**To Reproduce**
```py
import asyncio
import queue
from typing import Iterable

# partially unknown error is emitted
a1: asyncio.Future
a2: queue.Queue

# here not
b1: list[asyncio.Future]
b2: list[queue.Queue]

# for other generic classes there is an error
c1: list[list]
c2: list[Iterable]
```
```
error: Type of "Future" is partially unknown
    Type of "Future" is "Type[Future[Unknown]]" (reportUnknownMemberType)
error: Type of "Queue" is partially unknown
    Type of "Queue" is "Type[Queue[Unknown]]" (reportUnknownMemberType)
error: Expected type arguments for generic class "list" (reportMissingTypeArgument)
error: Expected type arguments for generic class "Iterable" (reportMissingTypeArgument)
```

**Expected behavior**
An error is emitted for `asyncio.Future`, `queue.Queue`, in `b1` and `b2` as well.

**VS Code extension or command-line**
v1.1.215

**Comment by erictraut** [comment]
I agree this is an inconsistency. There's some special-case logic to suppress this diagnostic when an unspecialized class is passed as an argument to a call expression. This logic has a bug and is also suppressing the diagnostic when an unspecialized class is used as a subscript within an index expression. A fix for this problem will be included in the next release.

**Comment by cdce8p** [comment]
Thanks! Tbh I wasn't sure initially if it was a pyright issue or something else as I couldn't reproduce it reliably. Glad to see it fixed.

**Comment by erictraut** [comment]
This is included in pyright 1.1.216, which I just published. It will also be included in the next release of pylance.
